agenta-backend/agenta_backend/services/db_manager.py

```diff
@@ -616,11 +618,35 @@ async def create_deployment(
             raise Exception(f"Error while creating deployment: {e}")
 
 
+async def get_app_type_from_template(template_id: Optional[str]) -> Enum:
+    """Get the application type from the specified template.
+
+    Args:
+        template_id (Optional[str]): The ID of the template
+
+    Returns:
+        AppType (Enum): The determined application type. Defaults to AppType.CUSTOM.
+    """
+
+    if template_id is None:
+        return AppType.CUSTOM
+
+    template_db = await get_template(template_id=template_id)
```

Comment:
So it looks like it's about `CUSTOM` vs `TEMPLATE_RAG` vs `TEMPLATE_GENERATION` vs `TEMPLATE_CONVERSATION`.

Reply:
I got the picture now. This is clear.